modules/broker-auth/src/main/java/io/ballerina/messaging/broker/auth/authentication/jaas/UserStoreLoginModule.java

```diff
@@ -72,60 +75,54 @@ public boolean login() throws LoginException {
         }
         userName = userNameCallback.getName();
         password = passwordCallback.getPassword();
-        success = validateUserPassword(userName, password);
+        AuthResult authResult = userStoreConnector.authenticate(userName, password);
+        success = authResult.isAuthenticated();
+        if (success) {
+            authenticationId = authResult.getUserId();
+        }
         return success;
     }
 
-    /**
-     * Authenticate user credentials using user-store manager
-     *
-     * @param userName Username
-     * @param password Password
-     * @return Whether user authentication success ot not
-     * @throws BrokerAuthException Throws if error occurred while authenticating user
-     */
-    private boolean validateUserPassword(String userName, char... password) throws BrokerAuthException {
-        return userName != null
-                && password != null
-                && userStoreManager.authenticate(userName, password);
-    }
-
     @Override
     public boolean commit() throws LoginException {
         if (success) {
-            UserPrincipal userPrincipal = new UserPrincipal(userName);
+            userPrincipal = new UserPrincipal(authenticationId);
             if (!subject.getPrincipals().contains(userPrincipal)) {
                 subject.getPrincipals().add(userPrincipal);
             }
-            userName = null;
-            if (password != null) {
-                for (int i = 0; i < password.length; i++) {
-                    password[i] = ' ';
-                }
-                password = null;
-            }
-            return true;
-        } else {
-            return false;
         }
+        cleanAuthInputData();
+        return success;
     }
 
     @Override
     public boolean abort() throws LoginException {
-        logout();
+        if (success) {
+            logout();
+        } else {
+            success = false;
```

Comment:
Do we need to set false? It should be false since we are in the else block. Isn't it?